band_together/entities/player/player.gd

```diff
@@ -190,3 +211,23 @@ func _on_dash_execute_timer_timeout() -> void:
 func _on_win_area_body_entered(_body: Node2D) -> void:
 	print("You Win!\n")
 	SceneTransition.change_scene("res://scenes/states/win/win.tscn")
+
+#Re-disables the attack hitbox after the agreed upon duration
+func _on_drum_timer_timeout() -> void:
+	$DrumArea/DrumAttack.disabled = true
+
+#Re-disables the attack hitbox after the agreed upon duration
+func _on_baton_timer_timeout() -> void:
+	$BatonArea/BatonAtack.disabled = true 
+
+##Consequence for enemies hitting the attack hitbox
+#Currently, as there is no health system for enemies, this rotates them :D
+func _on_drum_area_body_entered(body: Node2D) -> void:
+	if body.name != "Player":
+		body.rotate(1)
+
+##Consequence for enemies hitting the attack hitbox
+#Currently, as there is no health system for enemies, this rotates them :D
+func _on_baton_area_body_entered(body: Node2D) -> void:
+	if body.name != "Player":
+		body.rotate(1)
```

Comment:
Just a thought but maybe at one point, we could even separate the logic for movement and attacks. 🤔